Pinta.Effects/Effects/VoronoiDiagramEffect.cs

```diff
@@ -0,0 +1,284 @@
+			or ColorSorting.VerticalGBR
+			or ColorSorting.VerticalGRB
+			or ColorSorting.VerticalRBG
+			or ColorSorting.VerticalRGB => basePoints.OrderBy (p => p.Y).ThenBy (p => p.X),
+
+			_ => throw new InvalidEnumArgumentException (
+				nameof (colorSorting),
+				(int) colorSorting,
+				typeof (ColorSorting)),
+		};
+
+	private static Func<PointI, PointI, double> GetDistanceCalculator (DistanceCalculationMethod distanceCalculationMethod)
+	{
+		return distanceCalculationMethod switch {
+			DistanceCalculationMethod.Euclidean => Euclidean,
+			DistanceCalculationMethod.Manhattan => Manhattan,
+			DistanceCalculationMethod.Chebyshev => Chebyshev,
+			_ => throw new InvalidEnumArgumentException (
+				nameof (distanceCalculationMethod),
+				(int) distanceCalculationMethod,
+				typeof (DistanceCalculationMethod)),
+		};
+
+		static double Euclidean (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return difference.Magnitude ();
+		}
+
+		static double Manhattan (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return Math.Abs (difference.X) + Math.Abs (difference.Y);
+		}
+
+		static double Chebyshev (PointI targetPoint, PointI pixelLocation)
+		{
+			PointI difference = pixelLocation - targetPoint;
+			return Math.Max (Math.Abs (difference.X), Math.Abs (difference.Y));
+		}
+	}
+
+	private static ImmutableHashSet<PointI> CreatePoints (RectangleI roi, int pointCount, RandomSeed pointLocationsSeed)
+	{
+		int effectivePointCount = Math.Min (pointCount, roi.Width * roi.Height);
+
+		Random randomPositioner = new (pointLocationsSeed.Value);
+		var result = ImmutableHashSet.CreateBuilder<PointI> (); // Ensures points' uniqueness
+
+		while (result.Count < effectivePointCount) {
+
+			PointI point = new (
+				X: randomPositioner.Next (roi.Left, roi.Right + 1),
+				Y: randomPositioner.Next (roi.Top, roi.Bottom + 1)
+			);
+
+			result.Add (point);
+		}
+
+		return result.ToImmutable ();
+	}
+
+	private static IEnumerable<ColorBgra> CreateColors (int colorCount, RandomSeed colorsSeed)
+	{
+		Random randomColorizer = new (colorsSeed.Value);
+		HashSet<ColorBgra> uniquenessTracker = new ();
+		while (uniquenessTracker.Count < colorCount) {
+			ColorBgra candidateColor = randomColorizer.RandomColorBgra ();
+			if (uniquenessTracker.Contains (candidateColor)) continue;
+			uniquenessTracker.Add (candidateColor);
+			yield return candidateColor;
+		}
+	}
+
+	public sealed class VoronoiDiagramData : EffectData
+	{
+		[Caption ("Distance Calculation Method")]
+		public DistanceCalculationMethod DistanceCalculationMethod { get; set; } = DistanceCalculationMethod.Euclidean;
+
+		[Caption ("Point Count"), MinimumValue (1), MaximumValue (1024)]
+		public int PointCount { get; set; } = 100;
+
+		// Translators: The user can choose whether or not to render the points used in the calculation of a Voronoi diagram
+		[Caption ("Show Points")]
```

Comment:
Are there good use cases for this to be exposed to users? Seems more like a debugging option to me at first glance

Reply:
It could be an educational tool, so that they see that the cells are centered around the points, and then it piques their curiosity and then they look it up